packages/cli/src/gemini.test.tsx

```diff
@@ -21,7 +21,31 @@ import {
 } from './gemini.js';
 import { type LoadedSettings } from './config/settings.js';
 import { appEvents, AppEvent } from './utils/events.js';
-import type { Config } from '@google/gemini-cli-core';
+import { type Config } from '@google/gemini-cli-core';
+
+let onRenderCallback:
+  | ((renderMetrics: { renderTime: number }) => void)
+  | undefined;
+
+vi.mock('@google/gemini-cli-core', async (importOriginal) => {
+  const actual =
+    await importOriginal<typeof import('@google/gemini-cli-core')>();
+  return {
+    ...actual,
+    recordSlowRender: vi.fn(),
+  };
+});
+
+vi.mock('ink', async (importOriginal) => {
```

Comment:
nit: it would be better to not fake out handling of onRenderCallback.
Doing it this way makes the tests not representative of the real behavior of the Ink library. For example, now strangely onRender is basically never called and called at different times then in the real system. instead I would recommend using the vuilt in vi functionality to use fake time extended to mock out datetime.now and performance.now so that the ink code successfully uses the fake times.